### PR TITLE
Add tests for showing a CompositeException

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: test/distributed_exec.jl [hunk 1]
```diff
@@ -661,7 +661,20 @@ catch ex
     @test typeof(ex.exceptions[1].ex) == ErrorException
     errors = map(x->x.ex.msg, ex.exceptions)
     @test collect(1:5) == sort(map(x->parse(Int, x), errors))
-end
```
Comment (amitmurthy) [inline test/distributed_exec.jl]: You can remove the above 2 lines since they are anyway being tested below now.

File: test/distributed_exec.jl [hunk 2]
```diff
@@ -661,7 +661,20 @@ catch ex
...
+    # test start, next, and done
+    for (i, i_ex) in enumerate(ex)
+        @test i == parse(Int, i_ex.ex.msg)
+    end
+    # test showerror
+    err_strs = split(sprint(showerror, ex), "\n")
```
Comment (amitmurthy) [inline test/distributed_exec.jl]: Wouldn't it be simpler to replace this block with  
```
@test sprint(showerror, ex) == sprint(showerror, ex.exceptions[1]) * "\n\n...and 4 more exception(s).\n"
```
This way if the call stack changes, we do not have to come back and change the test here.

Comment (kshyatt) [inline test/distributed_exec.jl]: I suppose, but the line is super long then

Comment (amitmurthy) [inline test/distributed_exec.jl]: Hmm, you can split it over 3 lines then. Have `err_str`, `err_one_str` (without the `split`) and then have the test

**Comment by kshyatt** [comment]
Thanks for the feedback, @amitmurthy! Did I address everything?

**Review by amitmurthy — APPROVED** [review]

**Pull request merged**